src/__tests__/defaultRenderer.test.ts

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { ReactHooksRenderer } from '../types/react'
+
+describe('default renderer', () => {
+  beforeEach(() => {
+    jest.resetModules()
+  })
+
+  test('should resolve native renderer as default renderer', () => {
+    const expectedRenderer = require('../native/pure') as ReactHooksRenderer
+    const actualRenderer = require('..') as ReactHooksRenderer
+
+    expect(actualRenderer).toEqual(expectedRenderer)
+  })
+
+  test('should resolve dom renderer as default renderer', () => {
+    jest.doMock('react-test-renderer', () => {
+      throw new Error('missing dependency')
+    })
+
+    const expectedRenderer = require('../dom/pure') as ReactHooksRenderer
+    const actualRenderer = require('..') as ReactHooksRenderer
```

Comment:
had to use `require` in these tests otherwise the imports have resolved before the mocking occurs

Reply:
This is really interesting, I wasn't sure how to do this myself.